<!-- -*- mode: XML -*- -->
<!--
 This work is sponsored by the Department of the Air Force under Air Force
 Contract FA8721-05-C-0002. Opinions, interpretations, conclusions, and
 recommendations are those of the author and are not necessarily endorsed by
 the United States Government.
-->

<launch>
  <arg name="run_camera" default="true"/>
   <arg name="waypoint_check_distance" default="0.4"/>

  <include file="$(find racecar)/launch/includes/joy_estop.launch.xml" />
  <include file="$(find racecar)/launch/includes/steering.launch.xml" />
  <include file="$(find racecar)/launch/includes/throttle.launch.xml" />
  <include file="$(find racecar)/launch/includes/sensors.launch.xml">
    <arg name="run_camera" value="$(arg run_camera)" />
  </include> 
  
  <node pkg="rosbag" type="record" name="bag_recorder"
	args="record -O /media/DATA/bags/last_run.bag -a" /> 
	
  <include file="$(find racecar)/launch/mapping.launch" />
  
  <node pkg="racecar" type="laser_to_estop" name="laser_to_estop" output="screen">
    <param name="filter_alpha" value="0.95" />
    <param name="max_min_dist" value="0.2" />
    <param name="max_impact_time" value="0.9" />
  </node>
  
  <node pkg="racecar" type="goal_marker_node" name="goal_marker_node" output="screen"/>
  
  <node pkg="racecar" type="waypoint_tracker" name="waypoint_tracker">
    <param name="filter_alpha" value="0.8" />
    <param name="frequency" value="100.0" />
    <param name="speed_max" value="1.5" />
    <param name="throttle_max" value="0.95" />
    <param name="throttle_min" value="0.75" />
    <param name="throttle_dist_gain" value="1.0" />
    <param name="throttle_speed_gain" value="1.5" />
    <param name="steering_gain" value="1.0" />
    <param name="waypoint_check_distance" value="$(arg waypoint_check_distance)" />
  </node>

<node pkg="racecar" type="planner" name="planner" output="screen">
    <param name="tree_depth" value="8" />
    <param name="nb_obstacle_check" value="1000" />
    <param name="epsilon" value="0.9999" />
    <param name="inflation_radius" value="0.5" />
    <param name="waypoint_check_distance" value="$(arg waypoint_check_distance)" />
</node>

</launch>
